test/clojerl_Agent_SUITE.erl

```diff
@@ -0,0 +1,322 @@
+-module(clojerl_Agent_SUITE).
+
+-include("clojerl.hrl").
+-include("clojerl_int.hrl").
+-include("clj_test_utils.hrl").
+
+-export([ all/0
+        , init_per_suite/1
+        , end_per_suite/1
+        ]).
+
+-export([ close/1
+        , deref/1
+        , error_handler/1
+        , equiv/1
+        , meta/1
+        , restart/1
+        , release_pending/1
+        , str/1
+        , validator/1
+        , complete_coverage/1
+        ]).
+
+-spec all() -> [atom()].
+all() -> clj_test_utils:all(?MODULE).
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) -> clj_test_utils:init_per_suite(Config).
+
+-spec end_per_suite(config()) -> config().
+end_per_suite(Config) -> Config.
+
+%%------------------------------------------------------------------------------
+%% Test Cases
+%%------------------------------------------------------------------------------
+
+-spec close(config()) -> result().
+close(_Config) ->
+  0     = ets:info(?AGENT_TABLE, size),
+  Count = length(erlang:processes()),
+
+  Agent = 'clojerl.Agent':?CONSTRUCTOR(1),
+
+  1     = ets:info(?AGENT_TABLE, size),
+  Count = length(erlang:processes()) - 1,
+
+  ?NIL  = 'clojerl.Agent':close(Agent),
+
+  ok    = timer:sleep(100),
```

Comment:
I couldn't think of a better way of waiting for the process and the cleanup to be done.